### ISSUE TITLE
Issues with triggers section after react conversion

### ISSUE THREAD

**Issue by ezimanyi** [spinnaker/spinnaker#4423]
After the react conversion in spinnaker/deck#6970, the following triggers are not loading:
* Pubsub
* Wercker
* Webhook

Just selecting them from the drop-down in the trigger configuration screen causes a JavaScript error and the trigger fields never load:
![image](https://user-images.githubusercontent.com/1391982/57941260-784c2c00-789c-11e9-9bce-e8ccfaab686e.png)

If you open a pipeline that already has a trigger of one of the affected types, the entire section fails to load (including the drop-down to select a trigger type):
![image](https://user-images.githubusercontent.com/1391982/57941352-b34e5f80-789c-11e9-85f5-7b8064b635a3.png)

```
The JavaScript error in the PubSub case is:
PubsubTrigger.tsx:53 Uncaught TypeError: Cannot read property 'state' of undefined
    at PubSubTriggerContents (PubsubTrigger.tsx:53)
    at renderWithHooks (react-dom.development.js:12938)
    at mountIndeterminateComponent (react-dom.development.js:15020)
    at beginWork (react-dom.development.js:15625)
    at performUnitOfWork (react-dom.development.js:19312)
    at workLoop (react-dom.development.js:19352)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
    at replayUnitOfWork (react-dom.development.js:18578)
react-dom.development.js:17117 The above error occurred in the <PubSubTriggerContents> component:
    in PubSubTriggerContents (created by PubsubTrigger)
    in BaseTrigger (created by PubsubTrigger)
    in PubsubTrigger
angular.js:14961 TypeError: Cannot read property 'state' of undefined
    at PubSubTriggerContents (PubsubTrigger.tsx:53)
    at renderWithHooks (react-dom.development.js:12938)
    at mountIndeterminateComponent (react-dom.development.js:15020)
    at beginWork (react-dom.development.js:15625)
    at performUnitOfWork (react-dom.development.js:19312)
    at workLoop (react-dom.development.js:19352)
    at renderRoot (react-dom.development.js:19435)
    at performWorkOnRoot (react-dom.development.js:20342)
    at performWork (react-dom.development.js:20254)
    at performSyncWork (react-dom.development.js:20228) undefined
```

**Comment by ezimanyi** [comment]
@Jammy-Louie :  Mind taking a look into this?  I just ran into this today running of off `master` and confirmed by rolling back that this was introduced by the react conversion.

**Comment by ezimanyi** [comment]
Looks like this was addressed in spinnaker/deck#7024
